### ISSUE TITLE
[Migration] Migration probably shouldn't call migration script for current schema

### ISSUE THREAD

**Issue by avano** [syndesisio/syndesis#5364]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
When doing the upgrade from `1.6.9` to `latest`, I am on `28` schema from 1.6.9, however the migration script for schema `28` is called during the upgrade:

```
07:38:23.766 Current database schema version: 28
07:38:23.799 Migrating to schema: 28
Migration to schema 28 ...
This migration will auto add split steps to integrations with implicit split configured
Migrating integration ':i-LeuOQBxh0156frqpBxYz' - adding explicit split step
Number of integrations migrated: 1
Migrated to schema 28 completed
```

**Comment by heiko-braun** [comment]
@zregvart @christophd Any thoughts on this?

**Comment by avano** [comment]
What happened was that I created my integration on 7.3 version and I decided to not use the split step - I just used `SQL -> log`, however, during the upgrade the migration script for `28` schema was called and it added the split step in the integration

**Comment by zregvart** [comment]
Sounds like an easy fix to change `<=` to `<` here:

https://github.com/syndesisio/syndesis/blob/53f47a5a4db73ce55b9ea3a9fb3febb9e74b5687/app/server/runtime/src/main/java/io/syndesis/server/runtime/Migrations.java#L93

**Comment by avano** [comment]
imo rather start from version + 1 here :-)

https://github.com/syndesisio/syndesis/blob/720dbc5e32b200299535bcfe0cafb74883d6c9d8/app/server/cli/src/main/java/io/syndesis/server/cli/command/migrate/MigrateCommand.java#L100-L106

**Comment by zregvart** [comment]
You're absolutely right on that @avano :+1:

**Comment by heiko-braun** [comment]
@avano Mind a PR for this change?

**Comment by zregvart** [comment]
Closed via #5367
